src/main/java/de/tum/in/www1/artemis/service/plagiarism/ProgrammingPlagiarismDetectionService.java

```diff
@@ -343,41 +348,62 @@ public List<ProgrammingExerciseParticipation> filterStudentParticipationsForComp
                 }).toList();
     }
 
-    private List<Repository> downloadRepositories(ProgrammingExercise programmingExercise, List<ProgrammingExerciseParticipation> participations, String targetPath) {
+    private Optional<Repository> cloneTemplateRepository(ProgrammingExercise programmingExercise, String targetPath) {
+        try {
+            var templateRepo = gitService.getOrCheckoutRepository(programmingExercise.getTemplateParticipation(), targetPath);
+            gitService.resetToOriginHead(templateRepo); // start with clean state
+            return Optional.of(templateRepo);
+        }
+        catch (GitException | GitAPIException ex) {
+            log.error("Clone template repository {} in exercise '{}' did not work as expected: {}", programmingExercise.getTemplateParticipation().getVcsRepositoryUrl(),
+                    programmingExercise.getTitle(), ex.getMessage());
+            return Optional.empty();
+        }
+    }
+
+    private boolean shouldAddRepo(ProgrammingExercise programmingExercise, int minimumSize, Repository repo, Optional<Repository> templateRepo) {
+        if (templateRepo.isEmpty()) {
+            return true;
+        }
+
+        var diffToTemplate = programmingExerciseGitDiffReportService.calculateNumberOfDiffLinesBetweenRepos(programmingExercise.getTemplateParticipation().getVcsRepositoryUrl(),
```

Comment:
This is shorter and does the same thing:
```suggestion
        var diffToTemplate = programmingExerciseGitDiffReportService.calculateNumberOfDiffLinesBetweenRepos(getTemplateRepositoryUrl(),
```

Reply:
Thanks